src/rpcq.lisp

```diff
@@ -533,78 +210,197 @@ We distinguish between the following options for any field type:
     tbl))
 
 
-(defmacro defmessage (class-name field-specs &key (documentation nil))
-  "Create a (de-)serializable message definition with name CLASS-NAME and slots (SLOT-SPECS)."
+(defmacro defmessage (class-name parent-name field-specs &key (documentation nil))
+  "Create a (de-)serializable message definition.
+
+PARAMETERS:
+  * CLASS-NAME: The name of the message type.
+  * PARENT-NAMES: A list of length at most 1 of \"parent message types\". Child message types inherit their parents slot specifications.
```

Comment:
Mega nitpick: the parameter in the signature is `parent-name` vs `PARENT-NAMES` in the docs

Reply:
This is typical Lisp docstring convention: the parameter in the signature is aktschually recapitalized as PARENT-NAMES by the reader, and so if you do `(describe 'defmessage)` you'll see it as being capitalized. To wit:

```
CL-USER> (describe 'rpcq:defmessage)
RPCQ:DEFMESSAGE
  [symbol]

DEFMESSAGE names a macro:
  Lambda-list: (CLASS-NAME RPCQ::PARENT-NAME RPCQ::FIELD-SPECS &KEY
                           (DOCUMENTATION NIL))
  Documentation:
    Create a (de-)serializable message definition.
    
    PARAMETERS:
      * CLASS-NAME: The name of the message type.
...
```

Reply:
Oh, i meant singular vs plural, is that also automatically handled? That would seem like some black magic...